src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -105,12 +108,17 @@ private FileSmoosher(
 
   public void add(File fileToAdd) throws IOException
   {
-    add(fileToAdd.getName(), Files.map(fileToAdd));
+    add(fileToAdd.getName(), fileToAdd);
   }
 
   public void add(String name, File fileToAdd) throws IOException
   {
-    add(name, Files.map(fileToAdd));
+    MappedByteBuffer mappedFileBuffer = Files.map(fileToAdd);
+    try {
+      add(name, mappedFileBuffer);
+    } finally {
+      ByteBufferUtils.unmap(mappedFileBuffer);
```

Comment:
Yes... also there is #46, and also there is an idea to approach this more fundamentially with ResourceHandlers, discussed in https://github.com/druid-io/druid/pull/3422. Another option is to move away from ByteBuffers, to e. g. Netty's, Aeron's or Chronicle's alternatives, which are better in many ways. So I close this PR for now anyway.